commit 2a0cdf46753664293081d50e2bc2c46fe3b4c575 (refs/changes/37/2537/1)
Author: Todd Whiteman <todd.whiteman@joyent.com>
Date:   2017-09-09T14:29:06-07:00 (2 years, 1 month ago)
    
    IMGAPI-644 the docker image cache must clear activated image entries

diff --git a/CHANGES.md b/CHANGES.md
index e51c40d..277d000 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -1,5 +1,9 @@
 # IMGAPI changelog
 
+## 4.0.8
+
+- IMGAPI-644 the docker image cache must clear activated image entries
+
 ## 4.0.7
 
 - IMGAPI-642 missing file0 from imgapi docker image causing container provision
diff --git a/lib/docker.js b/lib/docker.js
index 6aa866b..bbe173f 100644
--- a/lib/docker.js
+++ b/lib/docker.js
@@ -167,6 +167,10 @@ function dockerDownloadAndImportLayer(opts, callback) {
             log.debug({digest: digest, uuid: uuid},
                 'dockerDownloadAndImportImage: image layer already cached');
 
+            // IMGAPI-644 This cached image should not be activated.
+            assert.equal(newImage.activated, false,
+                'newImage.activated should be false');
+
             progressStatus('Download complete (cached)');
             callback(null, newImage);
             return;
@@ -648,6 +652,9 @@ function _dockerActivateImage(newImage, ctx, callback) {
             return;
         }
 
+        // Remove the image from the cache.
+        delete DOCKER_IMAGE_CACHE[newImage.uuid];
+
         if (newFileInfo) {
             resMessage({
                 type: 'progress',
diff --git a/package.json b/package.json
index f8bf4ae..21f37c1 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "imgapi",
   "description": "Image API to manage images for SDC 7",
-  "version": "4.0.7",
+  "version": "4.0.8",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
